src/utils/image.utils.ts

```diff
@@ -0,0 +1,192 @@
+ * @returns {Promise<IExtendedBlob>} A Promise that resolves with the compressed image as a Blob.
+ */
+export const compressImage = ({ src, filename, options }: TCompressImage): Promise<IExtendedBlob> => {
+    const { maxWidth = DEFAULT_IMAGE_WIDTH, quality = DEFAULT_IMAGE_QUALITY } = options || {};
+
+    return new Promise((resolve, reject) => {
+        const image = new Image();
+        image.src = src;
+        image.onload = () => {
+            const canvas = document.createElement("canvas");
+            const canvas_context = canvas.getContext("2d");
+            if (!canvas_context || !(canvas_context instanceof CanvasRenderingContext2D)) {
+                return reject(new Error("Failed to get 2D context"));
+            }
+
+            if (image.naturalWidth > maxWidth) {
+                const width = DEFAULT_IMAGE_WIDTH;
+                const scaleFactor = width / image.naturalWidth;
+                canvas.width = width;
+                canvas.height = image.naturalHeight * scaleFactor;
+            } else {
+                canvas.width = image.naturalWidth;
+                canvas.height = image.naturalHeight;
+            }
+
+            canvas_context.fillStyle = "transparent";
+            canvas_context.fillRect(0, 0, canvas.width, canvas.height);
+            canvas_context.save();
+            canvas_context.drawImage(image, 0, 0, canvas.width, canvas.height);
+
+            canvas.toBlob(
+                (blob) => {
+                    if (!blob) return;
+                    const modified_filename = filename.replace(/\.[^/.]+$/, ".jpg");
+                    const file: IExtendedBlob = new Blob([blob], { type: "image/jpeg" });
+                    file.lastModifiedDate = Date.now();
+                    file.name = modified_filename;
+                    resolve(file);
+                },
+                "image/jpeg",
+                quality,
+            );
+        };
+    });
+};
+
+/**
+ * Convert a File to a Base64 encoded image representation.
+ * @param {File} file - The File object to convert to Base64.
+ * @returns {Promise<TBase64Image>} A Promise that resolves with an object containing the Base64 image data and the filename.
+ */
+export const convertToBase64 = (file: File): Promise<TBase64Image> => {
+    return new Promise((resolve) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onloadend = () => {
+            resolve({
+                src: reader.result?.toString() || "",
+                filename: file.name,
+            });
+        };
+    });
+};
+
+/**
+ * Check if a given filename has a supported image format extension.
+ *
+ * @param {string} filename - The filename to check for a supported image format.
+ * @returns {boolean} True if the filename has a supported image format extension, false otherwise.
+ */
+export const isSupportedImageFormat = (filename: string) => /\.(png|jpg|jpeg|gif|pdf)$/gi.test(filename ?? "");
```

Comment:
Can split this to a constant? These formats actually set based on what our Backend support and i think it should be a constant. https://api.deriv.com/api-explorer#document_upload

![image](https://github.com/deriv-com/deriv-utils/assets/75345074/94fab2f3-b3e4-491e-a80e-52a3cb1299a9)

Reply:
@yashim-deriv 
i added document format constants, pls review